api/src/main/java/uk/gov/hmcts/payment/api/controllers/OrderController.java

```diff
@@ -0,0 +1,254 @@
+        @ApiResponse(code = 403, message = "Forbidden-Access Denied"),
+        @ApiResponse(code = 422, message = "Invalid or missing attribute"),
+        @ApiResponse(code = 404, message = "No Service found for given CaseType"),
+        @ApiResponse(code = 504, message = "Unable to retrieve service information. Please try again later"),
+        @ApiResponse(code = 500, message = "Internal Server")
+    })
+    @PostMapping(value = "/order")
+    @Transactional
+    public ResponseEntity<?> create(@Valid @RequestBody OrderDto orderDto, @RequestHeader(required = false) MultiValueMap<String, String> headers) {
+        return new ResponseEntity<>(orderDomainService.create(orderDto, headers), HttpStatus.CREATED);
+    }
+
+    @ApiOperation(value = "Create credit account payment", notes = "Create credit account payment")
+    @ApiResponses(value = {
+        @ApiResponse(code = 201, message = "Payment created"),
+        @ApiResponse(code = 400, message = "Payment creation failed"),
+        @ApiResponse(code = 403, message = "Payment failed due to insufficient funds or the account being on hold"),
+        @ApiResponse(code = 404, message = "Account information could not be found"),
+        @ApiResponse(code = 504, message = "Unable to retrieve account information, please try again later"),
+        @ApiResponse(code = 422, message = "Invalid or missing attribute"),
+        @ApiResponse(code = 412, message = "The order has already been paid"),
+        @ApiResponse(code = 417, message = "The order amount should be equal to order balance")
+    })
+    @PostMapping(value = "/order/{order-reference}/credit-account-payment")
+    @ResponseBody
+    @Transactional
+    public ResponseEntity<OrderPaymentBo> createCreditAccountPayment(@RequestHeader(value = "idempotency_key") String idempotencyKey,
+                                                                     @PathVariable("order-reference") String orderReference,
+                                                                     @Valid @RequestBody OrderPaymentDto orderPaymentDto) throws CheckDigitException, InterruptedException, JsonProcessingException {
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        Function<String, Optional<IdempotencyKeys>> getIdempotencyKey = idempotencyKeyToCheck -> idempotencyKeysRepository.findByIdempotencyKey(idempotencyKeyToCheck);
+
+        Function<IdempotencyKeys, ResponseEntity<?>> validateHashcodeForRequest = idempotencyKeys -> {
+
+            OrderPaymentBo responseBO;
+            try {
+                if (!idempotencyKeys.getRequest_hashcode().equals(orderPaymentDto.hashCodeWithOrderReference(orderReference))) {
+                    return new ResponseEntity<>("Payment already present for idempotency key with different payment details", HttpStatus.CONFLICT); // 409 if hashcode not matched
+                }
+                if (idempotencyKeys.getResponseCode() >= 500) {
```

Comment:
This IF statement seems a bit strange to me. Please explain a reason behind it

Reply:
This is to deserialise response for timeout scenarios as the payload is in plain string